File: exercice1/Stock.php
<?php
namespace raouf;
class Stock{
	private $article=[];
	public function __construct($articles=array()){
		foreach($articles as $art){
			array_push($this->article, $art);
		}
	}
	public function voir_stock(){
		foreach($this->article as $item){
			var_dump($item);
		}
	}
	public function get_article($ref){

		if(empty($this->article)){
			return false;
		}else{
			
			foreach($this->article as $item){

				if($item->get_nrf()===$ref){
					return $item;
				}
			}
			return false;
		}
	}
	public function add_article(raouf\Article $article){
		if($this->get_article($article->get_nrf())){
			return false;
		}else{
			array_push($this->article, $article);
			return true;
		}
	}
	public function remove_article(raouf\Article $article){
		if($this->get_article($article->get_nrf())){
			for($i=0;$i<count($this->article);$i++){
				if($this->article[$i]->get_nrf()===$article->get_nrf()){
					unset($this->article[i]);
					return true;
				}
			}
		}else{
			return false;
		}
	}
	public function update_article($ref,$article){
		if($this->get_article($ref)){
			for($i=0;$i<count($this->article);$i++){
				if($this->article[$i]->get_nrf()===$ref){
					$this->article[$i]=$article;
					return true;
				}
			}
		}else{
			return false;
		}
	}
	public function get_with_name($name){
		if(empty($this->article)){
			return false;
		}else{
		foreach($this->article as $item){
			if($item->get_name()===$name){
				return $item;
			}
		}
		return false;
		}
	}
	public function search_with_intervale($interval=array()){
		foreach($this->article as $item){
			if($item->get_price()>= $interval[0] AND $item->get_price()<= $interval[1]){
				echo $item->get_name().'   :'.$item->get_price();

			}
		}
	}
}
?>

File: exercice1/python/Stock.py
class Stock:
	tab_article=[]
	def __init__(self, articles=[]):
		for item in articles:
			self.tab_article.append(item)
	def voir_stock(self):
		for item in self.tab_article:
			print(item)
	def serachWithRef(self, ref):
		if self.tab_article.count==0:
			return False
		else:
			for item in self.tab_article:
				if item.get_ref()==ref:
					return item
			return False
	def addArticle(self, article):
		if self.serachWithRef(article.get_ref()):
			return False
		else:
			self.tab_article.append(article)
			return True
	def deleteArticle(self, ref):
		if self.serachWithRef(ref):
			self.tab_article.remove(self.serachWithRef(ref))
			return True
		else:
			return False
	def updateArticle(self, ref, article):
		if self.serachWithRef(ref):
			for item in self.tab_article:
				if item.get_ref()==ref:
					item=article
					break
			return True
		else:
			return False
	def searchWithName(self, name):
		for item in self.tab_article:
			if item.get_name()==name:
				return item
		return False
	def searchWithInterval(self, interval=()):
		for item in self.tab_article:
			if item.get_price() >= interval[0] and item.get_price() <= interval[1]:
				print(item)